pkg/prompb/types.pb.go

```diff
@@ -1807,6 +2101,40 @@ func (m *TimeSeries) Unmarshal(dAtA []byte) error {
 				return err
 			}
 			iNdEx = postIndex
+		case 3:
+			if wireType != 2 {
+				return fmt.Errorf("proto: wrong wireType = %d for field Exemplars", wireType)
+			}
+			var msglen int
+			for shift := uint(0); ; shift += 7 {
+				if shift >= 64 {
+					return ErrIntOverflowTypes
+				}
+				if iNdEx >= l {
+					return io.ErrUnexpectedEOF
+				}
+				b := dAtA[iNdEx]
+				iNdEx++
+				msglen |= int(b&0x7F) << shift
+				if b < 0x80 {
+					break
+				}
+			}
+			if msglen < 0 {
+				return ErrInvalidLengthTypes
+			}
+			postIndex := iNdEx + msglen
+			if postIndex < 0 {
+				return ErrInvalidLengthTypes
+			}
+			if postIndex > l {
+				return io.ErrUnexpectedEOF
+			}
+			m.Exemplars = append(m.Exemplars, Exemplar{})
```

Comment:
We should implement the same optimizations we do for samples (maybe separate commit same pr)
```suggestion
			if len(m.Examplars) < cap(m.Examplars) {
				m.Examplars = m.Examplars[:len(m.Examplars)+1]
				m.Examplars[len(m.Examplars)-1].Reset()
			} else {
				m.Examplars = append(m.Examplars, Examplars{})
			}
```
in custom.ts.go we need
```
func (m *TimeSeries) Reset() {
	*m = TimeSeries{Labels: m.Labels[:0], Samples: m.Samples[:0], Examplars: m.Exemplars[:0]}
}
```